workProj.py

```diff
@@ -1,8 +1,59 @@
-import re, csv
+import re, csv, os
+from datetime import datetime, date, time
 
-#1st step. Read the salesData.csv and turn the data into a Dict
+os.chdir("c:\\users\\pdinh\\python\\workproject") #for working in shell
+#CSV_FILENAME = "salesData.csv"
+#"os.getcwd() = {}", os.getcwd()
+#"__file__ = {}", __file__
+#"os.path.dirname(__file__) = {}", os.path.dirname(__file__)
+
+#fileLocalDir = os.path.dirname(__file__)
+#absoluteCsvFilePath = os.path.join(fileLocalDir, CSV_FILENAME)
+
+
+#1st step. Read the salesData.csv and turn the data into a Dict.
 myDict = {}
->>> with open("salesData.csv", newline="") as csvfile:
+with open("salesData.csv", newline="") as csvfile:
 	reader = csv.reader(csvfile)
 	for row in reader:
-		myDict[row[0]] = row[1]
\ No newline at end of file
+		myDict[row[0]] = row[1]
+
+#2nd step. Use regex to return a new Dict with the same keys but the values trimmed to be only Dates.
+def myDictRegex(dict):
```

Comment:
Best practice for a module that has functions and "main" execution is to have function definitions up top (underneath imports) and the "main" execution under all the functions.